File: exercises/java/src/test/java/tax/TaxCalculatorAfterFirstYearTest.java
package tax;

import org.junit.Before;
import org.junit.Ignore;
import org.junit.Test;

import java.time.LocalDate;

import static org.assertj.core.api.Assertions.assertThat;
import static java.time.Month.APRIL;
import static tax.FuelType.*;

@Ignore
public class TaxCalculatorAfterFirstYearTest {

    private static final LocalDate FIRST_OF_APRIL_2017 = LocalDate.of(2017, APRIL, 1);
    private TaxCalculator taxCalculator;

    @Before
    public void setUp() {
        taxCalculator = new DummyTaxCalculator();
    }

    @Test
    public void subsequentYearsTaxForPetrol() {
        Vehicle vehicle = new Vehicle(206, PETROL, FIRST_OF_APRIL_2017, 20000);
        assertThat(taxCalculator.calculateTax(vehicle)).isEqualTo(140);
    }

    @Test
    public void subsequentYearsTaxForElectric() {
        Vehicle vehicle = new Vehicle(206, ELECTRIC, FIRST_OF_APRIL_2017, 20000);
        assertThat(taxCalculator.calculateTax(vehicle)).isEqualTo(0);
    }

    @Test
    public void subsequentYearsTaxForAlternativeFuel() {
        Vehicle vehicle = new Vehicle(206, ALTERNATIVE_FUEL, FIRST_OF_APRIL_2017, 20000);
        assertThat(taxCalculator.calculateTax(vehicle)).isEqualTo(130);
    }
}

File: exercises/javascript/test/tax-calculator-after-first-year.test.js
const { DummyTaxCalculator } = require('./dummy-tax-calculator');
const { Vehicle } = require('../vehicle');
const { FuelType } = require('../fuel-type');

describe.skip('Tax calculator on vehicles after the first year', () => {
  let taxCalculator = new DummyTaxCalculator();
  let FIRST_OF_APRIL_2017 = new Date(2017, 4, 1);

  it('subsequent years tax for petrol', () => {
    const vehicle = new Vehicle(206, FuelType.PETROL, FIRST_OF_APRIL_2017, 20000);
    expect(taxCalculator.calculateTax(vehicle)).toBe(140);
  })

  it('subsequent years tax for electric', () => {
    const vehicle = new Vehicle(206, FuelType.ELECTRIC, FIRST_OF_APRIL_2017, 20000);
    expect(taxCalculator.calculateTax(vehicle)).toBe(0);
  })

  it('subsequent years tax for alternative fuel', () => {
    const vehicle = new Vehicle(206, FuelType.ALTERNATIVE_FUEL, FIRST_OF_APRIL_2017, 20000);
    expect(taxCalculator.calculateTax(vehicle)).toBe(130);
  })
})
